src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -206,3 +220,85 @@ private[hyperspace] object IndexSummary {
     indexDirPath.toString
   }
 }
+
+/**
+ * Case class representing index metadata and index statistics from latest index version.
+ *
+ * @param name Index name.
+ * @param indexedColumns Indexed columns.
+ * @param includedColumns Included columns.
+ * @param numBuckets Number of buckets.
+ * @param schema Index schema json.
+ * @param state Index state.
+ * @param kind Index kind.
+ * @param hasLineage Lineage enabled on index.
+ * @param numIndexFiles Total number of index files.
+ * @param sizeIndexFiles Total size of index files.
+ * @param numSourceFiles Total number of source data files.
+ * @param sizeSourceFiles Total size of source data files.
+ * @param numAppendedFiles Total number of appended source data files.
+ * @param sizeAppendedFiles Total size of appended source data files.
+ * @param numDeletedFiles Total number of deleted source data files.
+ * @param sizeDeletedFiles Total size of deleted source data files.
+ */
+private[hyperspace] case class IndexStatistics(
+    name: String,
+    indexedColumns: Seq[String],
+    includedColumns: Seq[String],
+    numBuckets: Int,
+    schema: String,
+    state: String,
+    kind: String,
+    hasLineage: Boolean,
+    numIndexFiles: Int,
+    sizeIndexFiles: Long,
+    numSourceFiles: Int,
+    sizeSourceFiles: Long,
+    numAppendedFiles: Int,
+    sizeAppendedFiles: Long,
+    numDeletedFiles: Int,
+    sizeDeletedFiles: Long,
+    indexRootPaths: Seq[String])
+
+private[hyperspace] object IndexStatistics {
+  def apply(spark: SparkSession, entry: IndexLogEntry): IndexStatistics = {
+    val indexSummary = IndexSummary(spark, entry)
+    IndexStatistics(
+      indexSummary.name,
+      indexSummary.indexedColumns,
+      indexSummary.includedColumns,
+      indexSummary.numBuckets,
+      indexSummary.schema,
+      indexSummary.state,
+      entry.derivedDataset.kind,
+      entry.hasLineageColumn,
+      entry.content.fileInfos.size,
+      entry.content.fileInfos.map(_.size).sum,
+      entry.sourceFileInfoSet.size,
+      entry.sourceFileInfoSet.map(_.size).sum,
+      entry.appendedFiles.size,
+      entry.appendedFiles.map(_.size).sum,
+      entry.deletedFiles.size,
+      entry.deletedFiles.map(_.size).sum,
+      getIndexRootPaths(entry))
+  }
+
+  /**
+   * Extract top-most index directories which contain index files. Each index directory
+   * contains index files created for an active version of index.
```

Comment:
it's for the "latest" version of index. I suggested to add this field and thought this would be helpful to show which (version of) index data is referred for the latest index. (so that users can delete outdated versions or perform optimizeIndex.